### PR TITLE
BENCH: Add benchmarks for morphology.watershed

### PR DESCRIPTION
**Pull request by lagru**
## Description
A prerequisite to #3233 ensuring that performance regressions are noticed. 

Some notes for anyone wondering:
- `data.coins()` should provide a more realistic benchmark than a synthetic image.
- I'm trying to use as much of the input handling provided by `watershed` as possible, therefore the arguments `markers` and `connectivity` are only provided as integers and not prepared ahead of time.
- This benchmark is not comprehensive (only 2d, not all parameters are used) but should cover most use cases.

## For reviewers
- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [x] <strike>Check that new functions are imported in corresponding `__init__.py`.</strike>
- [x] <strike>Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.</strike>
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

cc @jni, @soupault

### CODE REVIEW

File: benchmarks/morphology.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    params = [(5, 50, 500), (1, 2), (0, 0.01)]
+
+    def setup(self, *args):
+        self.image = data.coins()
+
+    def time_watershed(self, seed_count, connectivity, compactness):
+        morphology.watershed(self.image, seed_count, connectivity,
+                             compactness=compactness)
+
+    def mem_watershed(self, seed_count, connectivity, compactness):
```
Comment (jni) [inline benchmarks/morphology.py]: Sorry, this is actually incorrect (my bad in the template), see the correct way to measure memory usage in #3249

File: benchmarks/morphology.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+from skimage import data, morphology
+
+
+class Watershed(object):
+
+    param_names = ["seed_count", "connectivity", "compactness"]
+    params = [(5, 50, 500), (1, 2), (0, 0.01)]
+
+    def setup(self, *args):
+        self.image = data.coins()
```
Comment (jni) [inline benchmarks/morphology.py]: Let's apply `filters.sobel` to coins to make it a more realistic use case.

Comment (jni) [reply benchmarks/morphology.py]: Other than this small comment, this looks good to me!

**Comment by lagru** [comment]
@jni, @soupault Sorry, for bothering you. What's the status on this? From my perspective it looks like this one is ready to go.

File: benchmarks/benchmark_morphology.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import numpy as np
+from skimage import data, filters, morphology
+
+
+class Watershed(object):
+
+    param_names = ["seed_count", "connectivity", "compactness"]
+    params = [(5, 500), (1, 2), (0, 0.01)]
+
+    def setup(self, *args):
```
Comment (soupault) [inline benchmarks/benchmark_morphology.py]: Aren't `*args` here and below redundant?

**Comment by lagru** [comment]
> yes it's a tick from me! But fyi what I meant is that you could include this in #3233 -- @jni

Ah, sorry. I didn't understand it that way. I thought providing the benchmarks before actually making performance related changes would be the way to go. Although I recently realized asv can benchmark older commits without the new benchmarks just fine. 
So if you would prefer to have these changes into #3233 I'm happy to oblige.

**Comment by jni** [comment]
@lagru either is fine. Hopefully someone else will swoop in and merge this soon... =)

**Comment by soupault** [comment]
Thanks @lagru !

**Comment by lagru** [comment]
You're welcome.

**Pull request merged**